code/game/objects/items/devices/beacon.dm

```diff
@@ -20,14 +20,18 @@
 	GLOB.teleportbeacons -= src
 	return ..()
 
+/obj/item/beacon/proc/turn_off()
+	icon_state = "beacon-off"
+	GLOB.teleportbeacons -= src
+	src.weak_reference = null
```

Comment:
I don't think anything should be `null`ing its own weakref, this should be fixed in some other way

One idea: Send a signal in `/obj/machinery/teleport/hub/proc/teleport` to `target`, something like `COMSIG_MOVABLE_BEING_TELEPORTED_TO`, then have beacons hook the signal and return a blocking flag if disabled

Another idea, have beacons send a signal when turned off `COMSIG_BEACON_DISABLED` that teleport hubs can hook onto and clear the target if caught

Reply:
Yup, seems a lot better idea than just outright removing weak_ref. I'm going to implement it.